[winchus] Moving Winchus keyboard to release after revamping documentation.

alex-larkin opened the pull request:
* /experimental/w/winchus/ deleted and /release/w/winchus/ created
* Welcome.htm, winchus.php, and readme.md completely revised and reformatted. 
* Some other small tweaks and revisions made to keyboard such as showing diacritics on [K_COLON] on the onscreen keyboard, and a flick correction to the symbols layer of the touch layout. Winchus.kps package details description updated. 

Could the reviewer please check if my links to winchus.pdf (in winchus/source/help/) work in welcome.htm, winchus.php, and readme.md? Also, if my (#english) anchor link and comments in readme.md are working as expected. 

I welcome any feedback and will be happy to make any needed corrections.

Thanks so much!

keyman-server commented:
Thank you for your pull request. You'll see a "build failed" message until the Keyman team has reviewed the pull request and manually initiated the build process.

Every change committed to this branch will become part of this pull request. When you have finished submitting files and are ready for the Keyman team to review this pull request, please post a "Ready for review" comment.

alex-larkin commented:
Ready for review. Also, I meant to ask if the reviewer could please check if my Winchus.kps package details description is looking okay. Thanks!

darcywong00 commented on release/w/winchus/source/winchus.kps:
```diff
@@ -0,0 +1,230 @@
+      <FileType>.htm</FileType>
+    </File>
+  </Files>
+  <Keyboards/>
```
The keyboard package fails to build because you need to add the winchus keyboard to it.
In Keyman Developer, go to the package editor to add the keyboard.

Remember to also assign the Quechua language to the touch layout keyboard (.js)

alex-larkin replied:
Thanks! Just to be sure, what's the extension I should look for?

darcywong00 replied:
Add the files in the build/ folder:
* winchus.kmx
* winchus.js

darcywong00 commented on release/w/winchus/source/winchus.kps:
```diff
@@ -0,0 +1,230 @@
+      <CopyLocation>0</CopyLocation>
+      <FileType>.md</FileType>
+    </File>
+    <File>
+      <Name>welcome\winchus.pdf</Name>
+      <Description>File winchus.pdf</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.pdf</FileType>
```
A copy of this file needs to be in the welcome/ folder. (It's in the help/)

alex-larkin replied:
Good catch! I'll fix that

darcywong00 commented on release/w/winchus/source/help/winchus.php:
```diff
@@ -0,0 +1,2416 @@
+    <a href="winchus.pdf" style="text-decoration: none;">
+        <div style="border: 1px solid black; padding: 10px; display: inline-block; text-align: center; background-color: #f2f2f2;">
+            <b>Descarga este documento<br />para leerlo sin conexión</b>
+            <img src="file-type-pdf-47.png" style="display: block; margin-left: auto; margin-right: auto; padding-top: 10px;" />
```
I think all the assets for the php file are in the `images/` folder.

alex-larkin replied:
Thanks! Should I move the images up a level or change the link to images/filename.png?

darcywong00 replied:
Can leave the images and update all the links

darcywong00 reviewed (dismissed):
Some missing files and broken links

alex-larkin commented:
Thanks so much! I'll take a look at this on Monday afternoon Peru time,

DavidLRowe commented:
We would encourage you to remove the version number and date from the README.md file in order that, when/if the keyboard is updated, this README.md file will not need modification.

On Monday I'll try to look at the images, files and folders. I think when you create the PDF file it includes all referenced images, so those images don't need to be available for the PDF file to display properly. In contrast, the .php file (the online help for the keyboard) and the welcome.htm file (the help included with the keyboard package) need to have any referenced images available. For the .php file, that means they need to be in the "help" folder. For the welcome.htm file, they need to be included in the package's file list (in the .kps file).

Currently the package (.kps file, Keyboards tab) has no BCP 47 codes defined.
On the Details tab, please select welcome.htm for the Welcome file and readme.htm for the Readme file.

alex-larkin commented:
Hi there!

Thanks for all of the valuable feedback. I'm sorry these updates took a little longer than I had hoped. 

I made the changes requested:
* Pulled images out of /images directory and up into /help. (Is that okay? It will make updating the winchus.PHP from the welcome.htm a lot easier)
* Copied winchus.pdf over
* Added winchus.kmx and winchus.js
* I also spell-checked my comments in Winchus.kmn

How is it looking?

DavidLRowe commented:
> Pulled images out of /images folder and up into /help. (Is that okay? It will make updating the winchus.PHP from the welcome.htm a lot easier)

More that just "okay", I think moving the images out of the "images" folder into the "help" folder is the best solution for your very detailed documentation! However, from a quick look it seems that this change is not yet completed:

- It's true that the .php file (from what I can see) no longer refers to the "images" folder
- however the files themselves are still in the "images" folder and need to be moved.

You may need to delete all the files in the "images" folder on GitHub, then copy them from your computer to the "help" folder on GitHub. Once you've done that, GitHub will figure out that you're really doing a move.

DavidLRowe commented:
It looks as though the welcome.htm references the winchus.pdf file, so a copy of that file is needed in the "welcome" folder (where the package's file list correctly thinks it should be).

(The welcome.htm file and the files it references provide the help that is packaged with the keyboard. The winchus.php file and the files it references provide the help that is available online. A future enhancement should allow set of source files to provide both types of help files, but until then we need to have duplicate copies of the information.)

When I try to compile the package, it complains that there is no keyboard file and, indeed, in the "Files" list there is no reference to .kmx, .kvk or .js files. Those can be added by using the "Add..." button on that page. In addition, on the "Keyboards" tab, you should add one or more BCP 47 codes corresponding to languages the keyboard supports.

I wonder if the changes you noted above were actually made or are waiting to be completed?

alex-larkin commented:
🤔 Hmn, this is my first time trying to update a PR. In my [fork](https://github.com/alex-larkin/keyboards/tree/master/winchus) all the files are as they should be, and when I [committed](https://github.com/keymanapp/keyboards/pull/3354/commits/dc3a2c75a2917dd333e6d900b352f4cbec06533d) then I saw it updated here: 

![image](https://github.com/user-attachments/assets/a30ede57-e2e0-4784-b23e-f0e65334cf39)

Thoughts? Do I need to do something different?

Thanks!

DavidLRowe commented:
Unfortunately, your new changes ended up in the wrong place. (Sadly, I can attest that this is very easy to do!) Instead of being a sibling folder of "release", "winchus" needs to be a child folder of "w" which is a child folder of "release".
![image](https://github.com/user-attachments/assets/7e03359f-c70b-4fb7-93ed-0e43521ea451)

How to fix this? 
(First a side note: The recommended way to work is to create a "branch" (called whatever you like) where you place your changes and then the pull request is based on that. Any updates to that branch become part of the pull request. If you decide the pull request is in error or not needed, you can delete the branch with no consequences. In this case you've based the pull request on your "master" branch, so this option is not available, short of closing this PR, deleting your fork, reforking the keyboards repository and creating a new pull request, which would be perfectly fine if you think that would be easier.)

You need to delete the "winchus" folder that is at the root (sibling of "release") along with all the files and folders under it. Then you'll need to redo the changes you did above. If this isn't enough direction please write again.

alex-larkin commented:
Thank you David for helping me to sort that out. 

It looks like when I tried reuploading the /winchus folder, I dropped it in my forked root by mistake, instead of /w as you said. 

I deleted an extra branch from my fork, deleted /winchus from keymanapps/keyboards and reuploaded to keymanapps/keyboards/release/w/.

Then i deleted the /images folder found in /help. 

I think that now everything is where it should be. Please let me know how it's looking!

Thanks so much.

darcywong00 commented on release/w/winchus/source/help/winchus.php:
```diff
@@ -0,0 +1,2416 @@
+        border-bottom: 0px solid #B3B1B1;
+        border-radius: 25px;
+        color: #000000;
+    }
```
This might be a stray closing brace that needs to be removed

alex-larkin replied:
Looks like it's part of the .kbd-resultado class but was indented wrong. I fixed it.

darcywong00 commented on release/w/winchus/source/help/winchus.php:
```diff
@@ -0,0 +1,2416 @@
+          background-repeat: no-repeat;
+          width: 34px;
+        }
+        
+        .special {
+          background-repeat: no-repeat;
+          
+        }
+        
+        .keycap {
+          font: bold 7pt Arial; 
+          position: absolute; 
+          left: 6px; 
+          top: 6px;
+        }
+        
+        .key .keycap {          
+          display: none;
+        }
+        
+        .special .keycap {
+          display: block;
+        }
+        
+        .keytext {
+          
+					font:
+						12pt
+							 
+						"Arial"; 
+          position: absolute; 
+          display: block;
+          right: 5px;
+          bottom: 4px;
+          color: blue;
+        }
+
+        /* FIN DE CLASES PARA MOSTRAR EL TECLADO*/
+        
+        END;
```
This entire CSS block needs to go inside the `<?php` block near the top.

An example is arbore.php
```php
<?php
  $pagename = 'Arbore Keyboard Help';
  $pagetitle = $pagename;
  require_once('header.php');

  $pagestyle = <<<END
    .coldiv { display: flex; flex-direction: row; gap: 5%;}
    table, tr, th, td { font: 10pt Arial; border-collapse: collapse; border: solid 1px; padding: 5px 5px; text-align: center;}
    th { font-weight: bold;}
    .combo { font-family: Courier, monospace; padding: 0px 10px;}
    .empty { background-color: lightgray;}
  END;

?>
```

alex-larkin replied:
Ok. Maybe this could be clarified at https://help.keyman.com/developer/keyboards/phphelpfile ? Or maybe I could suggest an edit directly later today.

darcywong00 commented on release/w/winchus/source/help/winchus.php:
```diff
@@ -0,0 +1,2416 @@
+  $pagetitle = $pagename;
+  require_once('header.php');
+?>
+<meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Winchus</title>
```
The `<title>Winchus</title>` can be removed since it's handled by $pagetitle above

alex-larkin replied:
Ok. Maybe this could be clarified at https://help.keyman.com/developer/keyboards/phphelpfile ? Or maybe I could suggest an edit directly later today.

darcywong00 commented on release/w/winchus/source/help/winchus.php:
```diff
@@ -0,0 +1,2416 @@
+        /* FIN DE CLASES PARA MOSTRAR EL TECLADO*/
+        
+        END;
+</head>
+<body>
```
```suggestion
```

These aren't needed. The bulk of the php file is treated as body

alex-larkin replied:
Ok. Maybe this could be clarified at https://help.keyman.com/developer/keyboards/phphelpfile ? Or maybe I could suggest an edit directly later today.

DavidLRowe commented:
@alex-larkin The files and folders all look like they are in the right places. Congratulations! Please review the comments that @darcywong00 made. He's more familiar with the PHP format than I am, so I value his review. I think with those changes this pull request should be ready to merge.

darcywong00 commented on release/w/winchus/source/help/winchus.php:
```diff
@@ -2,8 +2,8 @@
-?>
-<meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Winchus</title>
+
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
```
I think you can remove this line too. (It's HTML that doesn't belong in the PHP block).

alex-larkin commented:
Ok, thanks. I just did that.

alex-larkin commented on release/w/winchus/source/help/winchus.php:
```diff
@@ -0,0 +1,2170 @@
+          display: block;
+        }
+        
+        .keytext {
+          
+					font:
+						12pt
+							 
+						"Arial"; 
```
Interesting. Actually, lines 165 through 239 are copied in from the .html file generated by Keyman Developer > [projectnamne].kmn > On-Screen > Export... and that's how Keyman Developer generates that class. 

Maybe that could be fixed there too? 

Actually, should this entire section could be deleted in the PHP file since the PHP file has a special way of handling this? ("Dynamically Constructed Keyboard Images" at https://help.keyman.com/developer/keyboards/phphelpfile)

alex-larkin commented:
🥳 Thanks so much David and Darcy!

DavidLRowe commented:
@alex-larkin This should be published within a few hours.

alex-larkin commented:
Awesome possum

alex-larkin commented:
For my next version should I delete the CSS pertaining to the Dynamically Constructed Keyboard Images? ([See this comment](#discussion_r1973746236))

LornaSIL commented:
Looks like the keyboard is already online.

You can go ahead and update the .php file in a new pull request. The online help file can have changes made without updating the keyboard version number since it's not part of the package. Also, please look at the help page: 
https://help.keyman.com/keyboard/winchus/1.1/winchus

You don't need the table of contents at all. The help file automatically generates a TOC on the right side, so it would be best to remove that cruft.
Also, there's tons of white space at the end that isn't needed. You could delete all those `<br/>` codes.
